### PR TITLE
Issue #1132: More efficient parsing of CSV and TSV input.

### PR DESCRIPTION
**Pull request by andrewruckerjones**
Signed-off-by: Andrew Rucker Jones <arjones@t-online.de>

This PR addresses GitHub issue: #1132.

Briefly describe the changes proposed in this PR:

* Implemented mapping strategy necessary to move TSV processing to opencsv to take advantage of opencsv's parallel processing.
* Moved duplicated code to a base class.
* Moved to version 4.3.2 of opencsv (up from 4.2).

### CODE REVIEW

**Comment by barthanssens** [comment]
Thanks, since it's an upgrade from 4.2 to 4.3.2, it requires approval from Eclipse, so I'll submit a CQ (should be a formality)

**Comment by barthanssens** [comment]
CQ 17911 submitted

File: queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVReader.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.eclipse.rdf4j.query.resultio.text.tsv;
```
Comment (jeenbroekstra) [inline queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVReader.java]: Editorial: class is missing a license header

File: queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVMappingStrategy.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
+package org.eclipse.rdf4j.query.resultio.text.tsv;
```
Comment (jeenbroekstra) [inline queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVMappingStrategy.java]: Class is missing a license header.

File: queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/SPARQLResultsXSVMappingStrategy.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
+package org.eclipse.rdf4j.query.resultio.text;
```
Comment (jeenbroekstra) [inline queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/SPARQLResultsXSVMappingStrategy.java]: Ditto: license header missing.

**Comment by jeenbroekstra** [comment]
@andrewruckerjones I rebased this PR to the develop branch (because it's not, strictly speaking, just a bug fix, so I'd prefer to not include it in patch release), but this introduced a conflict, possibly because your fork's branch is out of sync with the project's develop branch. I tried to fix by merging the latest master branch into your PR, but this doesn't seem to resolve it. 

If you have a minute, could you take a look at conflict and resolve it?

File: queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVReader.java (hunk 1 above)
Comment (andrewruckerjones) [reply queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVReader.java]: Added license header.

File: queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVMappingStrategy.java (hunk 2 above)
Comment (andrewruckerjones) [reply queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/tsv/SPARQLResultsTSVMappingStrategy.java]: Added license header.

File: queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/SPARQLResultsXSVMappingStrategy.java (hunk 3 above)
Comment (andrewruckerjones) [reply queryresultio/text/src/main/java/org/eclipse/rdf4j/query/resultio/text/SPARQLResultsXSVMappingStrategy.java]: Added license header.

**Comment by andrewruckerjones** [comment]
All three conflicts were resolved by taking this branch over develop because the code that was removed was moved to the (new) base class.

The IP validation check fails still, but I don't know how to fix that through the Github interface.

**Comment by jeenbroekstra** [comment]
@andrewruckerjones thanks for this! The IP validation fail is just triggered by the odd generated e-mail address on your merge commit - I can overrule this, so no problem :)

**Pull request merged**